replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicatorRunner.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.security.service.SecurityServiceException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.codice.ddf.security.common.Security;
+import org.codice.ditto.replication.api.ReplicationStatus;
+import org.codice.ditto.replication.api.Replicator;
+import org.codice.ditto.replication.api.ReplicatorConfigLoader;
+import org.codice.ditto.replication.api.SyncRequest;
+import org.codice.ditto.replication.api.impl.data.SyncRequestImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The ReplicatorRunner periodical queues up replication jobs for all the current replication
+ * configurations.
+ */
+public class ReplicatorRunner {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ReplicatorRunner.class);
+
+  private final Security security;
+
+  private final Replicator replicator;
+
+  private final ReplicatorConfigLoader replicatorConfigLoader;
+
+  private final ScheduledExecutorService scheduledExecutor;
+
+  public ReplicatorRunner(
+      ScheduledExecutorService scheduledExecutor,
+      Replicator replicator,
+      ReplicatorConfigLoader replicatorConfigLoader) {
+    this(scheduledExecutor, replicator, replicatorConfigLoader, Security.getInstance());
+  }
+
+  public ReplicatorRunner(
+      ScheduledExecutorService scheduledExecutor,
+      Replicator replicator,
+      ReplicatorConfigLoader replicatorConfigLoader,
+      Security security) {
+    this.scheduledExecutor = notNull(scheduledExecutor);
+    this.replicator = notNull(replicator);
+    this.replicatorConfigLoader = notNull(replicatorConfigLoader);
+    this.security = security;
+  }
+
+  public void init() {
+    long period =
+        Long.parseLong(
+            System.getProperty(
+                "org.codice.replication.period", String.valueOf(TimeUnit.MINUTES.toSeconds(5))));
+    scheduledExecutor.scheduleAtFixedRate(
+        this::replicateAsSystemUser, TimeUnit.MINUTES.toSeconds(1), period, TimeUnit.SECONDS);
```

Comment:
✏️ 1 minute should probably be defined as a constant for clarity

Reply:
Added constant